text/0104-string-parameterization-output-spec.md

```diff
@@ -0,0 +1,264 @@
+
+### Known Route Segments
+
+A parameterizer that has knowledge of the route structure that the URL corresponds to must replace URL parameters with the correct route segment names.
+
+For example, the React Router SDK has knowledge of the URL structure of the app. Therefore, a parameterizer in the SDK should refer to the route structure, and supply the names of the URL segments. The same applies to web frameworks like Django. It is also in theory possible for a parameterizer to cross-reference a URL to a known service like GitHub against GitHub’s documentation, to determine whether outgoing requests match a specific API route.
+
+**The parameterizer must replace route parameters with the name of the parameter surrounded in curly braces.**
+
+Given:
+
+```
+GET http://someservice.com/users/tom/
+```
+
+If the parameterizer determines that `tom` is a route parameter named `:name`, it must return:
+
+```
+GET http://someservice.com/users/{name}
+```
+
+**Note:** The parameterizer should replace the smallest possible substring. e.g., the URL `/service/id-17` should be parameterized as `/service/id-{user_id}` rather than `/service/{user_id}`.
+
+### Unknown Route Segments
+
+If a parameterizer does not have knowledge of the route structure, but it determines that a value within a URL is a dynamic segment, **it must replace the value with a single `*` character.**
+
+Given:
+
+```
+GET http://someservice.com/users/tom/info
+```
+
+If the parameterizer determines that `tom` is a parameter of unknown significance, it must return:
+
+```
+GET http://someservice.com/users/*/info
+```
+
+**Note:** The parameterizer must preserve the original URL structure as much as possible. e.g., the URL `https://service.io/data/sentry/frontend/info` where both `sentry` and `frontend` are dynamic route segments, the parameterizer must return `https://service.io/data/*/*/info` rather than `https://service.io/data/*/info`.
+
+### Subdomains
+
+Subdomain can contain parameters, too. For example, in the URL `https://company.someservice.io/` the string `company` might correspond to a client name, and is a dynamic string.
+
+If a parameterizer determines that a section of a domain string is dynamic, **it must replace the segment with a single `*` character.**
+
+Given:
+
+```
+GET http://host1.someservice.com/users/
+```
+
+If the parameterizer determines that `host1` is a parameter, it must return:
+
+```
+GET http://*.someservice.com/users/
+```
+
+**Note:** The parameterizer must not alter domains or hostnames in any other way.
+
+### Port Numbers
+
+**If the URL contains a port number, the parameterizer should not parameterize it.**
+
+Given:
+
+```
+GET http://someservice.com:8080/
+```
+
+Return:
+
+```
+GET http://someservice.com:8080/
+```
+
+### Auth Strings
+
+**If the URL contains authentication strings, the parameterizer must replace them with a `*` character.**
```

Comment:
[Advanced data scrubbing](https://docs.sentry.io/product/data-management-settings/scrubbing/advanced-datascrubbing/#methods) is a powerful feature Sentry offers to users to deal with sensitive data. Do we think it's a good idea to ignore this feature and redact the strings in a different way?

Reply:
Specifically, in advanced data scrubbing, the user can chose whether they want to [remove, mask, hash, or replace](https://docs.sentry.io/product/data-management-settings/scrubbing/advanced-datascrubbing/#methods) the original value (which is then documented in the `_meta` field, as @untitaker described [above](https://github.com/getsentry/rfcs/pull/104/files#r1236533531)).